src/main/java/com/vinylteam/vinyl/util/impl/JunoVinylParser.java

```diff
@@ -0,0 +1,138 @@
+package com.vinylteam.vinyl.util.impl;
+
+import com.vinylteam.vinyl.entity.Currency;
+import com.vinylteam.vinyl.entity.RawOffer;
+import com.vinylteam.vinyl.util.VinylParser;
+import lombok.extern.slf4j.Slf4j;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+import static java.util.stream.Collectors.toSet;
+
+@Slf4j
+public class JunoVinylParser implements VinylParser {
+
+    private final String startBaseLink = "https://www.juno.co.uk/all/";
+    private final String startLink = startBaseLink + "back-cat/2/?media_type=RawOffer";
+
+    private final String RawOfferItemsListSelector = "div.product-list";
+    private final String priceBlockSelector = "div.pl-buy";
+    private final String priceLineSelector = "span.price_lrg.text-cta";
+    private final String RawOfferItemSelector = "div.dv-item";
+    private final String imageLinkSelector = "img.lazy_img.img-fluid";
+    private final String RawOfferInfoBlockSelector = "div.pl-info";
+    private final String RawOfferInfoItemsSelector = "div.vi-text";
```

Comment:
1) All variables names from starts from lower case
2) We will have one instance of the class. We can make all variables static constants.